[CI] SmokeTestWatcherTestSuiteIT testMonitorClusterHealth failing

**Build Scans:**
- [elasticsearch-intake #39797 / part2](https://gradle-enterprise.elastic.co/s/j4ijzl3unur2o)
- [elasticsearch-pull-request #145335 / part-2](https://gradle-enterprise.elastic.co/s/am3lkgvdhtjxs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:watcher:javaRestTest" --tests "org.elasticsearch.smoketest.SmokeTestWatcherTestSuiteIT.testMonitorClusterHealth" -Dtests.seed=4D6BA118500D5E03 -Dtests.locale=ru-KZ -Dtests.timezone=Asia/Yakutsk -Druntime.java=26
```

**Applicable branches:**
9.4

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.smoketest.SmokeTestWatcherTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testMonitorClusterHealth'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Watch count (from _watcher/stats)
Expected: is <1>
     but: was <0>
```

**Issue Reasons:**
- [9.4] 1 consecutive failures in test testMonitorClusterHealth
- [9.4] 1 consecutive failures in step part2
- [9.4] 1 consecutive failures in pipeline elasticsearch-intake
- [9.4] 2 failures in test testMonitorClusterHealth (2.1% fail rate in 97 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -244,9 +244,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.ForecastIT
   method: testOverflowToDisk
   issue: https://github.com/elastic/elasticsearch/issues/148606
-- class: org.elasticsearch.smoketest.SmokeTestWatcherTestSuiteIT
-  method: testMonitorClusterHealth
-  issue: https://github.com/elastic/elasticsearch/issues/148615
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:UNMAPPED_FIELDS_LOAD}
   issue: https://github.com/elastic/elasticsearch/issues/148619

--- a/x-pack/plugin/watcher/src/javaRestTest/java/org/elasticsearch/smoketest/SmokeTestWatcherTestSuiteIT.java
+++ b/x-pack/plugin/watcher/src/javaRestTest/java/org/elasticsearch/smoketest/SmokeTestWatcherTestSuiteIT.java
@@ -107,8 +107,11 @@ public class SmokeTestWatcherTestSuiteIT extends WatcherRestTestCase {
             indexWatch(watchId, builder);
         }
 
-        // check watch count
-        assertWatchCount(1);
+        // check watch count — wrap in assertBusy because when the .watches index is created for the
+        // first time, the WatcherLifeCycleService detects the new shard allocation and triggers a
+        // reload (pauseExecution + async reloadInner). During the pause window the watch count is 0;
+        // it becomes 1 once reloadInner finishes loading watches from the index.
+        assertBusy(() -> assertWatchCount(1));
 
         // check watch history
         ObjectPath objectPath = getWatchHistoryEntry(watchId);

